packages/astro/src/core/compile/style.ts

```diff
@@ -31,13 +36,65 @@ export function createStylePreprocessor(
 			}
 
 			return { code: result.code, map };
-		} catch (err) {
-			errors.push(err as unknown as Error);
-			return {
-				error: err + '',
-			};
+		} catch (err: any) {
+			try {
+				err = enhanceCSSError(err, filename);
+			} catch {}
+			cssTransformErrors.push(err);
+			return { error: err + '' };
 		}
 	};
+}
+
+function enhanceCSSError(err: any, filename: string) {
```

Comment:
This function is largely copied over from @Princesseuh's CSS error handling previously in `packages/astro/src/vite-style-transform/style-transform.ts`